src/Sylius/Bundle/ApiBundle/Doctrine/QueryCollectionExtension/TaxonCollectionExtension.php

```diff
@@ -40,18 +40,24 @@ public function applyToCollection(
             return;
         }
 
-        Assert::keyExists($context, ContextKeys::CHANNEL);
-        $channelMenuTaxon = $context[ContextKeys::CHANNEL]->getMenuTaxon();
+        $rootAlias = $queryBuilder->getRootAliases()[0];
 
         $user = $this->userContext->getUser();
         if ($user instanceof AdminUserInterface && in_array('ROLE_API_ACCESS', $user->getRoles(), true)) {
+            $queryBuilder
+                ->addOrderBy(sprintf('%s.parent', $rootAlias))
+                ->addOrderBy(sprintf('%s.position', $rootAlias))
+            ;
```

Comment:
I'm not sure if this is the right approach to return the list of taxons in the admin, but I have no better idea how to arrange them by default 🤔 I appreciate all suggestions and comments 😃

Reply:
I have mixed feelings 😄 It seems reasonable at first sight, but it also looks strange a little bit. If I understand it correctly, if we have a taxon tree like this:

```
- T-Shirts
    - Man
    - Woman
- Caps
    - Winter
    - Summer
- Accessories
```

we will receive them in the endpoint (which is a flat list) as:

```
- T-Shirts
- Caps
- Accessories
- Man
- Woman
- Winter
- Summer
```

which does not seem nice 😅 Maybe if we want to have the flat list of taxons we should just order them by IDs? 🤔 and propose another endpoint with a tree-like response? that the proposals from the top of my head 🖖